activemodel/test/cases/errors_test.rb

```diff
@@ -31,30 +31,29 @@ def self.lookup_ancestors
   end
 
   def test_delete
-    errors = ActiveModel::Errors.new(self)
-    errors[:foo] << "omg"
-    errors.delete("foo")
-    assert_empty errors[:foo]
+    errors = ActiveModel::Errors.new(Person.new)
+    errors.add(:name)
+    errors.delete("name")
+    assert_empty errors[:name]
   end
 
   def test_include?
     errors = ActiveModel::Errors.new(self)
-    errors[:foo] << "omg"
+    errors.add(:foo, "omg")
```

Comment:
Is it possible to wrap `errors[:foo] << "omg"` with `assert_deprecated { }` and have the test still pass?

Reply:
@tenderlove Yes got it. However this particular case was one of the incompatibility that can' really be solved. This is an emulated hash and you mentioned about placing a freeze to it.

However it is possible if I return a subclass of Hash, and override `<<`. If you want this I will try implementing it.